### PR TITLE
[EPICSYSTEM-23] Add Public comment period feed.

### PR DESCRIPTION
**Pull request by tolkamps1**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
 		"@tanstack/react-query-devtools": "^5.24.1",
 		"axios": "^1.6.7",
 		"fast-deep-equal": "^3.1.3",
+		"moment-timezone": "^0.5.45",
```
Comment (rpyyj) [inline package.json]: Let's not use `moment` and `moment-timezone` as they are no longer maintained. Please use `date-fns` if you need to pull in date utilities. We may want to roll our own utilities (in the new date service!) if they are just date comparison and date math calls. Up to you.

File: src/components/PcpFeed/PcpCard/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		"& div": {
+			fontWeight: 700,
+			padding: "0.4rem 0.875rem 0.4rem 0.875rem",
+			borderRadius: "1rem",
+			fontSize: "0.875rem",
+			lineHeight: "1rem",
+			color: "white",
+			display: "flex",
+			justifyContent: "center",
+			width: "5.624rem",
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: typo?

File: src/components/PcpFeed/PcpCard/index.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	},
+	upcoming: {
+		background: "#FCBA19",
+	},
+}));
+
+const PcpCard = ({ dateCompleted, dateStarted, phaseName, projectName }) => {
+	const { classes } = useStyles();
+
+	const formattedDate = (date) => {
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: Please remove this and use `getLongDate` from the date service (coming soon in PR #1).

File: src/components/PcpFeed/PcpCard/index.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	};
+
+	const status = (startDate, endDate) => {
+		if (startDate && endDate) {
+			const now = new Date();
+			const dateStarted = moment(startDate);
+			const dateCompleted = moment(endDate);
+			if (moment(now).isBetween(dateStarted, dateCompleted)) {
+				return "Open";
+			} else if (moment(now).isAfter(dateCompleted)) {
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: The `else`s are not necessary:
```
if (moment(now).isBetween(dateStarted, dateCompleted)) {
	return "Open";
}
if (moment(now).isAfter(dateCompleted)) {
	return "Closed";
}
return "Upcoming";
```

File: src/components/PcpFeed/PcpCard/index.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+			day: "numeric",
+		}).format(new Date(date));
+	};
+
+	const status = (startDate, endDate) => {
+		if (startDate && endDate) {
+			const now = new Date();
+			const dateStarted = moment(startDate);
+			const dateCompleted = moment(endDate);
+			if (moment(now).isBetween(dateStarted, dateCompleted)) {
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: Please use `date-fns` equivalents or add a js Date functions to the date service for these comparisons.

File: src/components/PcpFeed/PcpFooter.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+const useStyles = makeStyles()((theme) => ({
+	container: {
+		color: "white",
+		display: "flex",
+		padding: "0.625rem 1rem 0.625rem 0",
+		width: "100%",
+	},
+	pcpButton: {
+		fontSize: 20,
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpFooter.js]: Please use `rem` units

File: src/components/PcpFeed/PcpHeader.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+		padding: "4rem, 0.625rem, 0.625rem, 0",
+		justifyContent: "space-between",
+		width: "100%",
+		flexDirection: "column",
+	},
+	headerTitle: {
+		fontWeight: "700",
+		fontSize: "1.5rem",
+		color: "black",
+		margin: ".625rem 0 0 0",
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpHeader.js]: Please change to `"0.625rem 0 0 0"`

File: src/components/PcpFeed/index.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+	content: {
+		padding: "0 2rem 0 1rem",
+	},
+}));
+
+const PcpFeed = ({ setSelectedTab }) => {
+	const { classes } = useStyles();
+
+	const now = new Date();
+	const thirtyDaysAgo = moment(now).subtract(30, "days").format("YYYY-MM-DD");
```
Comment (rpyyj) [inline src/components/PcpFeed/index.js]: Same idea with this `moment` call

**Review by rpyyj — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (tolkamps1) [reply package.json]: Removed in 0ac6fed

File: src/components/PcpFeed/PcpCard/index.js (hunk 2 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: No, this is based on the figma where the status badge is a fixed size. Should this be done differently?

File: src/components/PcpFeed/PcpFooter.js (hunk 6 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpFooter.js]: Addressed in c53b5c2

File: src/components/PcpFeed/PcpHeader.js (hunk 7 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpHeader.js]: Addressed in c53b5c2

File: src/components/PcpFeed/index.js (hunk 8 above)
Comment (tolkamps1) [reply src/components/PcpFeed/index.js]: Addressed in c53b5c2

File: src/components/PcpFeed/PcpCard/index.js (hunk 3 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: Addressed in c53b5c2

File: src/components/PcpFeed/PcpCard/index.js (hunk 4 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: Addressed in c53b5c2

File: src/components/PcpFeed/PcpCard/index.js (hunk 5 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: Addressed in c53b5c2

File: src/components/PcpFeed/PcpCard/index.js (hunk 2 above)
Comment (rpyyj) [reply src/components/PcpFeed/PcpCard/index.js]: No, no. It's just that 5.624 is not a "16th" number, and I assumed you meant 5.625rem (90px).

I just checked the figma and I think the width there is 75px?

File: src/components/PcpFeed/PcpCard/index.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+		flexDirection: "row",
+		justifyContent: "space-between",
+		flexWrap: "wrap",
+	},
+	status: {
+		display: "inline-flex",
+		marginLeft: "auto",
+		"& div": {
+			fontWeight: 700,
+			padding: "0.4rem 0.875rem 0.4rem 0.875rem",
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: And I'm just curious why 0.4rem? 😄

File: src/components/PcpFeed/PcpCard/index.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
+import PropTypes from "prop-types";
+import { makeStyles } from "tss-react/mui";
+
+import { getFormatDateLongMonth, isAfterDate, isBetweenDates } from "services/date";
+
+const useStyles = makeStyles()((theme) => ({
+	container: {
+		color: theme.palette.text.primary,
+		background: "white",
+		padding: "0.6rem 0 0.6rem 0",
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: I'm just curious why 0.6rem?

File: src/components/PcpFeed/index.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+		padding: "0 2rem 0 1rem",
+	},
+}));
+
+const PcpFeed = ({ setSelectedTab }) => {
+	const { classes } = useStyles();
+
+	const now = new Date();
+	const thirtyDaysAgo = getFormatDateAsISOString(subtractDays(now, 30));
+	console.log(thirtyDaysAgo);
```
Comment (rpyyj) [inline src/components/PcpFeed/index.js]: Please remove the `console.log`

File: src/components/PcpFeed/PcpCard/index.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+				return "Open";
+			}
+			if (isAfterDate(dateCompleted)) {
+				return "Closed";
+			}
+			return "Upcoming";
+		}
+	};
+
+	const pcpStatus = status(dateStarted, dateCompleted);
```
Comment (rpyyj) [inline src/components/PcpFeed/PcpCard/index.js]: Please pull the `status` function code into here and wrap it with `useMemo`.

File: src/services/date.js [hunk 13]
```diff
@@ -7,3 +7,30 @@ export const getFormatDateLongMonth = (date) => {
 
 	return formattedDate;
 };
+
+export const getFormatDateAsISOString = (date) => {
```
Comment (rpyyj) [inline src/services/date.js:11]: This is a little simpler: `date.toISOString().substring(0, 10)`

**Review by rpyyj — CHANGES_REQUESTED** [review]

File: src/services/date.js (hunk 13 above)
Comment (tolkamps1) [reply src/services/date.js:11]: updated in 18da33b

File: src/components/PcpFeed/PcpCard/index.js (hunk 12 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: updated in 18da33b

File: src/components/PcpFeed/index.js (hunk 11 above)
Comment (tolkamps1) [reply src/components/PcpFeed/index.js]: updated in 18da33b

File: src/components/PcpFeed/PcpCard/index.js (hunk 10 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: Whoops, all my rem stuff got messed up. updated in 18da33b

File: src/components/PcpFeed/PcpCard/index.js (hunk 9 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: Updated in 18da33b

File: src/components/PcpFeed/PcpCard/index.js (hunk 2 above)
Comment (tolkamps1) [reply src/components/PcpFeed/PcpCard/index.js]: oh haha yea, the figma doesn't have "Upcoming" which is longer than the other status', but the weird rem was a mistake. Updated in 18da33b.

**Review by rpyyj — APPROVED** [review]

**Pull request merged**